thyrel-api/test_thyrel_api/PlayerDataProviderTest.cs

```diff
@@ -2,20 +2,89 @@
 using System.Threading.Tasks;
 using NUnit.Framework;
 using thyrel_api.DataProvider;
+using thyrel_api.Models;
 
 namespace test_thyrel_api
 {
     public class PlayerDataProviderTest : TestProvider
     {
         private IPlayerDataProvider _playerDataProvider;
-        
+
         [SetUp]
         public async Task Setup()
         {
             await SetupTest();
             _playerDataProvider = new PlayerDataProvider(Options);
         }
 
+        [Test]
+        public async Task Add()
+        {
+            var username = "Tintin";
+            var avatarUrl = " ";
+            var isOwner = true;
+            var roomId = 1;
+            var tokenId = 1;
+
+            var playerCount = Context.Player.Count();
+            await _playerDataProvider.Add(username, avatarUrl, isOwner, roomId, tokenId);
+            Assert.AreEqual(playerCount + 1, Context.Player.Count());
+        }
+
+        [Test]
+        public async Task GetPlayer()
+        {
+            var playerId = 1;
+            var player = Context.Player.First(p => p.Id == playerId);
+
+            var playerFromProvider = await _playerDataProvider.GetPlayer(player.Id);
+            Assert.AreEqual(player.Id, playerFromProvider.Id);
+            var playerNull = await _playerDataProvider.GetPlayer(-1);
+            Assert.IsNull(playerNull);
+        }
+
+        [Test]
+        public async Task GetPlayerByToken()
+        {
+            var playerId = 1;
+            var player = Context.Player.First(p => p.Id == playerId);
+            var token = new Token("xxxTokenKeyxxx");
+            player.Token = token;
+            await Context.SaveChangesAsync();
+
+            var playerFromToken = await _playerDataProvider.GetPlayerByToken(token.TokenKey);
+            Assert.AreEqual(player.Id, playerFromToken.Id);
+        }
+
+        [Test]
+        public async Task TestDisableNotExistingPlayerShouldReturnNull()
+        {
+            Assert.IsNull(await _playerDataProvider.Disable(-1));
+        }
+        [Test]
+        public async Task TestDisablePlayer()
+        {
+            var playerId = 1;
+            var player = Context.Player.First(p => p.Id == playerId);
+            var disabledPlayer = await _playerDataProvider.Disable(player.Id);
+            Assert.IsNotNull(disabledPlayer.DisableAt);
+        }
+        [Test]
+        public async Task TestSetOwnerOfNotExistingPlayerShouldReturnNull()
+        {
+            Assert.IsNull(await _playerDataProvider.SetOwner(0, true));
+        }
+        
+        [Test]
+        public async Task TestSetOwner()
+        {
+            var playerId = 1;
+            var player = Context.Player.First(p => p.Id == playerId);
```

Comment:
```suggestion
            var player = Context.Player.First(p => !p.IsOwner);
```
en moins tu es sur de tomber sur un player qui n'est pas déjà owner dans quelle cas dans test n'est pas signficatif.

Reply:
Okay c'est juste j'avais pas pensé!
j'attends que quelqu'un d'autre review pour changer cette ligne, comme ça si jamais il y a des autres choses à changer je ferais un seul commit de refractoring 
:)

Reply:
top parfait spam alex mdrrr